app/models.py

```diff
@@ -0,0 +1,244 @@
+                                back_populates="users")
+
+    @property
+    def password(self):
+        """Raise error when reading a User password attribute."""
+        raise AttributeError('password is not a readable attribute')
+
+    @password.setter
+    def password(self, password):
+        """Convert passwords to hashed values."""
+        self.password_hash = generate_password_hash(password)
+
+    def verify_password(self, password):
+        """Verify passwords with the hashed equivalent."""
+        return check_password_hash(self.password_hash, password)
+
+    def save(self):
+        """Save a user to database."""
+        db.session.add(self)
+        db.session.commit()
+
+    def generate_auth_token(self, expiration):
+        """Generate a random token with an expiration time."""
+        s = Serializer(current_app.config['SECRET_KEY'],
+                       expires_in=expiration)
+        return s.dumps({'id': self.id})
+
+    @staticmethod
+    def verify_auth_token(token):
+        """Verify token."""
+        s = Serializer(current_app.config['SECRET_KEY'])
+        try:
+            data = s.loads(token)
+        except SignatureExpired:
+            return None  # valid token, but expired
+        except BadSignature:
+            return None  # invalid token
+        user = User.query.get(data['id'])
+        return user
+
+    @login_manager.user_loader
+    def load_user(user_id):
+        """Return a user object if available or None otherwise."""
+        return User.query.get(int(user_id))
+
+    def __repr__(self):
+        """Represent objects with a readable format."""
+        return "<User(email='%s')>" % self.email
+
+
+class AnonymousUser(AnonymousUserMixin):
+    """Create an User with with AnonymousUserMixin properties."""
+
+    @staticmethod
+    def create_anonymous_user():
+        """Create anonymous user."""
+        anonymous_user = User(first_name='AnonymousUser',
+                              last_name='AnonymousUser', email='AnonymousUser')
+        anonymous_user.save()
+        return anonymous_user
+
+
+class ShortUrl(db.Model):
+    """Map ShortUrl class to short_url table in the database."""
+
+    __tablename__ = 'short_url'
+
+    id = db.Column(db.Integer, primary_key=True)
+    short_url = db.Column(db.String, unique=True)
+    is_active = db.Column(db.Boolean, default=True)
+    no_of_visits = db.Column(db.Integer, default=0)
+    deleted = db.Column(db.Boolean(), default=False, index=True)
+    when = db.Column(db.DateTime, default=db.func.current_timestamp(),
+                     nullable=False)
+    user_id = db.Column(db.Integer, db.ForeignKey('users.id'), nullable=False)
+    user = db.relationship("User", back_populates="short_urls")
+    long_url_id = db.Column(db.Integer, db.ForeignKey('long_url.id'),
+                            nullable=False)
+    long_url = db.relationship("LongUrl", back_populates="short_urls")
```

Comment:
My understanding is that the `user` property needs the foreign key column to be defined in other to return any user object associated with the short_url. I think it's the `user` property that is expendable here but that will mean writing more code to query any associated user using the foreign key column. I am yet to see any associative table without a foreign key column.